Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -30,217 +25,82 @@ internal func docConfigFilter(forNamespace namespace: MongoNamespace,
-        }
-
-        required init(namespace: MongoNamespace,
-                      documentId: HashableBSONValue,
-                      lastUncommittedChangeEvent: ChangeEvent<Document>?,
-                      lastResolution: Int64,
-                      lastKnownRemoteVersion: Document?,
-                      isStale: Bool,
-                      isPaused: Bool) {
-            self.namespace = namespace
-            self.documentId = documentId
-            self.uncommittedChangeEvent = lastUncommittedChangeEvent
-            self.lastResolution = lastResolution
-            self.lastKnownRemoteVersion = lastKnownRemoteVersion
-            self.isStale = isStale
-            self.isPaused = isPaused
-        }
-
-        static func == (lhs: CoreDocumentSynchronization.Config,
-                        rhs: CoreDocumentSynchronization.Config) -> Bool {
-            return lhs.documentId == rhs.documentId
-        }
-
-        func hash(into hasher: inout Hasher) {
-            documentId.hash(into: &hasher)
-        }
-
+final class CoreDocumentSynchronization: Codable, Hashable {
+    enum CodingKeys: String, CodingKey {
+        // These are snake_case because we are trying to keep the internal
+        // representation consistent across platforms
+        case namespace = "namespace", documentId = "document_id",
+        uncommittedChangeEvent = "last_uncommitted_change_event",
+        lastResolution = "last_resolution", lastKnownRemoteVersion = "last_known_remote_version",
+        isStale = "is_stale", isPaused = "is_paused", schemaVersion = "schema_version"
+        case docsColl = "docs_coll"
     }
 
     /// The collection we are storing document configs in.
-    private let docsColl: ThreadSafeMongoCollection<CoreDocumentSynchronization.Config>
+    private let docsColl: ThreadSafeMongoCollection<CoreDocumentSynchronization>
     /// Standard read-write lock.
-    private lazy var docLock = ReadWriteLock(label: "document_lock_\(namespace)_\(documentId.value)")
+    private let docLock: ReadWriteLock
     /// The error listener to propogate errors to.
     private weak var errorListener: FatalErrorListener?
-    /// The configuration for this document.
-    private(set) var config: Config
     /// The namespace this document is stored in.
-    var namespace: MongoNamespace { return config.namespace }
+    let namespace: MongoNamespace
     /// The id of this document.
-    var documentId: AnyBSONValue { return config.documentId.bsonValue }
+    let documentId: HashableBSONValue
 
     /// The most recent pending change event
-    var uncommittedChangeEvent: ChangeEvent<Document>? {
-        get {
-            return docLock.read {
-                return config.uncommittedChangeEvent
-            }
-        }
-        set(value) {
-            docLock.write {
-                self.config.uncommittedChangeEvent = value
-            }
-        }
-    }
-
+    var uncommittedChangeEvent: ChangeEvent<Document>?
```

Comment:
[minor] the semantics of this (as well as of `lastResolution` and `lastKnownRemoteVersion` have changed slightly. We no longer readlock the `docLock` when accessing this. I know it would be complicated to do so since we don't have a re-entrant lock but I think we need to make sure we're holding a readlock on the document whenever we're accessing these fields to be completely thread-safe